src/coreComponents/fileIO/vtk/VTKPolyDataWriterInterface.cpp

```diff
@@ -1156,14 +1156,24 @@ void VTKPolyDataWriterInterface::writeVtmFile( integer const cycle,
     {
 
       if( meshLevel.isShallowCopy() )
-      {
         return;
+
+      string const & meshLevelName = meshLevel.getName();
+
+      if( !m_levelNames.empty())
+      {
+        if( m_levelNames.find( meshLevelName ) == m_levelNames.end())
+          return;
       }
```

Comment:
```suggestion
      if( m_levelNames.find( meshLevelName ) == m_levelNames.end())
        return;
```

Reply:
Oh yes, you are right, sorry about that.
Can you then update the comments, descriptions and doxygens by adding something like "_Empty set means that all the level will be saved_" or something like that?
I'm not a huge fan of having `empty` means `all` because it means that the interface can't be easily used programmatically, but I guess we can live with this for this case here!